src/clj/puppetlabs/http/client/async.clj

```diff
@@ -242,7 +152,7 @@
    * :query-params - used to set the query parameters of an http request"
   [opts :- common/RawUserRequestOptions
    callback :- common/ResponseCallbackFn
-   client :- HttpAsyncClient]
+   client :- common/Client]
```

Comment:
Looks like the only places where this is currently used are here and in `create-default-client` in async.clj but `common/Client` is defed to `CloseableHttpAsyncClient`.  I do think it would be better to use the interface, `HttpAsyncClient`, rather than the class, `CloseableHttpAsyncClient` wherever we can.  I'd vote for either changing the definition of `common/Client` to use `HttpAsyncClient` or just doing away with `common/Client` completely and referencing `HttpAsyncClient` where needed.